app/src/main/java/com/bashkevich/androidfundamentals/moviesdetails/viewmodel/MoviesDetailsViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package com.bashkevich.androidfundamentals.moviesdetails.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.bashkevich.androidfundamentals.data.JsonLoad
+import com.bashkevich.androidfundamentals.data.Movie
+import kotlinx.coroutines.launch
+
+class MoviesDetailsViewModel(
+        private val jsonLoad: JsonLoad
+) : ViewModel() {
+    private var _movieLiveData = MutableLiveData<Movie>()
+
+    val movieLiveData: LiveData<Movie>
+        get() = _movieLiveData
+
+    fun getMovieFromList(movieId: Int) {
+        viewModelScope.launch {
```

Comment:
Since you already added loading of the list data only when it is empty. Could you also add here check that movie is loaded and movies id is same before you load it from the json?

Reply:
if (_movieLiveData.value?.id != movieId) that wiil be ok?